### ISSUE TITLE
ValueError when accessing Interval.fields after accessing Interval.attrs

### ISSUE THREAD

**Issue by blaiseli** [daler/pybedtools#251]
I came upon what looks a mysterious bug to me when using pybedtools on a bam file:
```
In [1]: filename = "../RNA_Seq_analyses/results_44hph_vs_38hph/hisat2/mapped_C_elegans/WT_44hph_4_polyU_on_C_elegans_sorted.bam"

In [2]: from pybedtools import BedTool

In [3]: bamfile = BedTool(filename)

In [4]: interval = next(iter(bamfile))

In [5]: interval
Out[5]: Interval(I:3754-3829)

In [6]: interval.fields
Out[6]: 
['NB501291:61:H3TLWBGX2:2:21111:16382:1546',
 '0',
 'I',
 '3755',
 '60',
 '1S73M1S',
 '*',
 '0',
 '0',
 'TGACCCAAAATTAAGCCCGCGAAGGCATGACTTCAGCGCGAGGCACTAGTTTCCAGAAGAACTCTGACGTCTACA',
 'AAAA/EEA6EEE///AA//A/E/E//AE/EA///E///E/E6/////A///<E/E/E//EEE///E//E/6A/A/',
 'AS:i:-14\tXN:i:0\tXM:i:4\tXO:i:0\tXG:i:0\tNM:i:4\tMD:Z:11C18G13G20T7\tYT:Z:UU\tNH:i:1']

In [7]: interval.attrs
Out[7]: {}

In [8]: interval.fields
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-8-be2d987bacca> in <module>()
----> 1 interval.fields

~/.local/lib/python3.6/site-packages/pybedtools/cbedtools.pyx in pybedtools.cbedtools.Interval.fields.__get__ (pybedtools/cbedtools.cxx:5613)()

~/.local/lib/python3.6/site-packages/pybedtools/cbedtools.pyx in pybedtools.cbedtools.Interval.deparse_attrs (pybedtools/cbedtools.cxx:5456)()

ValueError: Interval.attrs was not None, but this was a non-GFF Interval

```

This does not look like a desired behaviour: if `fields` was accessible once, shouldn't it be accessible again?

The same thing happens with a bed file:

```
In [9]: filename2 = "/Genomes/C_elegans/Caenorhabditis_elegans/Ensembl/WBcel235/Annotation/Genes/protein_coding.bed"

In [10]: bedfile = BedTool(filename2)

In [11]: interval2 = next(iter(bedfile))

In [12]: interval2
Out[12]: Interval(I:4118-10230)

In [13]: interval2.fields
Out[13]: ['I', '4118', '10230', 'WBGene00022277', '.', '-']

In [14]: interval2.attrs
Out[14]: {}

In [15]: interval2.fields
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-15-e67de7dc5926> in <module>()
----> 1 interval2.fields

~/.local/lib/python3.6/site-packages/pybedtools/cbedtools.pyx in pybedtools.cbedtools.Interval.fields.__get__ (pybedtools/cbedtools.cxx:5613)()

~/.local/lib/python3.6/site-packages/pybedtools/cbedtools.pyx in pybedtools.cbedtools.Interval.deparse_attrs (pybedtools/cbedtools.cxx:5456)()

ValueError: Interval.attrs was not None, but this was a non-GFF Interval

```

I'm working on Xubuntu 16.04 with a locally-compiled python 3.6.4, and with pybedtools version as follows:
```
In [16]: from pybedtools import __version__

In [17]: __version__
Out[17]: '0.7.10'

```

**Comment by daler** [comment]
Thanks for reporting, this is definitely a bug. At first glance I'm not exactly sure why this is happening, I'll look into it more.

**Comment by blaiseli** [comment]
I don't really understand the code, but it seems to happen during the call to `deparse_attrs` when `__get__`ting the `fields` property after `__get__`ting the `attrs` property in the following pieces of code from `cbedtools.pyx`:

```
    cpdef deparse_attrs(self):

        if self._attrs is None: return

        if self.file_type != "gff":
            raise ValueError('Interval.attrs was not None, but this was a non-GFF Interval')

        s = self._attrs.__str__()
        self._bed.fields[8] = _cppstr(s)
```

```
    property fields:
        def __get__(self):
            self.deparse_attrs()
            items = []
            for i in self._bed.fields:
                if isinstance(i, int):
                    items.append(i)
                else:
                    items.append(_pystr(i))
            return items
```

and

```
    property attrs:
        def __get__(self):
            if self._attrs is None:
                ft = _pystr(self._bed.file_type)
                if ft == 'gff':
                    self._attrs = Attributes(_pystr(self._bed.fields[8]))
                else:
                    self._attrs = Attributes("")
            return self._attrs
```

When dealing with a non-gff file type, `self._attrs = Attributes("")` results in a non-None `attrs`.

**Comment by daler** [comment]
Hi @blaiseli, this should now be fixed in #264
